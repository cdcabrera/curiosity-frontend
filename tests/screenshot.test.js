// const jestPlaywright = require('jest-playwright-preset');
// import { Browser, Page } from 'playwright';
const routesConfig = require('../src/config/routes');
const { routes: rt } = routesConfig;
// const { Browser } = require('playwright');
// Needs to be higher than the default Playwright timeout
// jest.setTimeout(40 * 1000);
// global['jest-playwright'] = {};

// const { chromium } = require('playwright'); // Or 'firefox' or 'webkit'.
// let browser;
// let page;

// (async () => {
// browser = await chromium.launch();
// page = await browser.newPage();
// await page.goto('https://example.com');
// other actions...
// await browser.close();
// })();

describe('Screenshot confirmations', () => {
  const routes = [...rt.map(({ path }) => path)];
  // let page;
  // beforeEach(async () => {
  //  await jestPlaywright.resetPage();
  //  await page.goto('https://whatismybrowser.com/');
  // });

  // beforeEach(async () => {
  // page = await browser.newPage();
  // });

  /*
  beforeAll(async () => {
    await page.goto('https://localhost:3000/insights/subscriptions/');
  });

  beforeEach(() => {
    // await jestPlaywright.resetBrowser();
  });

  // it.playwright('should have consistent product displays', async ({ page }) => {
  it('should have consistent product displays', async ({ page }) => {
    // const browser = Browser();
    // const page =
    console.log('>>>>>>>>>>', routes);
    console.log('>>>>>>>>>>', page);
    await page.goto('/rhel');
    await expect(page).toHaveScreenshot('/rhel');
  });
  */

  /*
  beforeAll(async () => {
    await page.goto('https://whatismybrowser.com/');
  });

  // test('should display correct browser', async () => {
  test('should display correct browser', async () => {
    // await page.goto('https://whatismybrowser.com/');
    const browser = await page.$eval('.string-major', el => el.innerHTML);
    expect(browser).toContain('Chrome');
  });
  */
  // let browser;
  // let page;

  beforeAll(async () => {
    // browser = await chromium.launchPersistentContext('./.chromium', {
    /*
    browser = await chromium.launchPersistentContext('', {
      ignoreHTTPSErrors: true,
      // baseURL: 'http://localhost:3000/insights/subscriptions/',
      baseURL: 'http://localhost:3000',
      headless: true,
      timeout: 40000
    });
    */
    // await new Promise(resolve => setTimeout(resolve, 30000));
  });
  afterAll(async () => {
    // await browser?.close();
  });
  beforeEach(async () => {
    // page = await browser.newPage();
    // page.once('load', () => console.log('Page loaded!'));
    // await page.reload();
  });
  afterEach(async () => {
    // await page?.close();
  });

  it('should display correct browser', async () => {
    // const browser = await chromium.launch();
    // const page = await browser.newPage();

    // console.log('>>>>>>>>>>>>>>>>>>>>', page);
    // await page.goto('http://localhost:3000/insights/subscriptions/');
    // const result = await page.$eval('.string-major', el => el.innerHTML);
    // expect(result).toContain('Chrome');

    await page.goto('http://localhost:3000/insights/subscriptions/rhel');
    // await page.goto('/insights/subscriptions/rhel');
    // await page.reload();
    // await page.waitForURL('/insights/subscriptions/rhel', { waitUntil: 'networkidle' });
    // await page.locator('.curiosity').waitFor();
    // const result = await page.$eval('main.subscriptions', el => el.innerHTML);
    const result = await page.$eval('.curiosity', el => el.innerHTML);
    expect(result).toMatchSnapshot('toolbarFieldUsage html');
  });
});
