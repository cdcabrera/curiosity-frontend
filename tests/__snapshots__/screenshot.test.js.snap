// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screenshot confirmations should display correct browser: toolbarFieldUsage html 1`] = `"<section class=\\"pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light\\" widget-type=\\"InsightsPageHeader\\"><div class=\\"pf-l-flex pf-m-justify-content-space-between-on-sm\\"><div class=\\"\\"><h1 data-ouia-component-type=\\"PF4/Title\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Title-3\\" widget-type=\\"InsightsPageHeaderTitle\\" class=\\"pf-c-title pf-m-2xl pf-u-mb-sm\\"><div class=\\"pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm\\"><span class=\\"pf-u-screen-reader\\"></span></div></h1></div></div></section><section class=\\"pf-c-page__main-section pf-m-no-padding curiosity \\"></section>"`;
